types/encoding.go

```diff
@@ -355,6 +479,62 @@ func (t Transaction) MarshalSia(w io.Writer) error {
 	return nil
 }
 
+// MarshalSiaSize returns the encoded size of t.
+func (t Transaction) MarshalSiaSize() (size int) {
+	size += 8
+	for _, sci := range t.SiacoinInputs {
+		size += len(sci.ParentID)
+		size += sci.UnlockConditions.MarshalSiaSize()
+	}
```

Comment:
I tried manually unrolling these like so:
```go
size += len(t.SiacoinInputs) * len(SiacoinOutputID{})
for _, sci := range t.SiacoinInputs {
    size += sci.UnlockConditions.MarshalSiaSize()
}
// etc.
```
but there was no difference in the benchmarks. It seems that the compiler optimizes this sort of code well.